src/App.tsx

```diff
@@ -1,24 +1,99 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useRef, useState } from 'react';
+import { Todo } from './types/Todo';
+import { TodosContext } from './components/TodosContext';
+import { TodosFilter } from './types/TodosFilter';
+import { TodosHeader } from './components/TodosHeader';
+import { TodosList } from './components/TodosList';
+import { getTodos } from './api/todos';
+import { TodosError } from './components/TodosError';
+import { TodosFooter } from './components/TodosFooter';
 
-const USER_ID = 0;
+const USER_ID = 11891;
+
+const DEFAULT_DATA = {
+  userId: USER_ID,
+  title: '',
+  completed: false,
+};
+
+const useFilter = (todos: Todo[], filter: string) => {
+  return todos.filter(todo => {
+    switch (filter) {
+      case TodosFilter.active:
+        return !todo.completed;
+      case TodosFilter.completed:
+        return todo.completed;
+      default:
+        return true;
```

Comment:
You can move switch out of the filter callback to avoid calling it if the case is default